### PR TITLE
feat(modal): add componentProps, so it's possible to pass data into t…

### LINKED ISSUE #261: [Enhancement] Modal: Add support for componentProps
**Describe the solution you'd like**
Currently the Kirby modal doesn't support data being passed to it when it is opened. It would be nice to be able to pass some context/componentProps.
The Ionic modal supports componentProps, and we are passing ModalWrapperConfig in that place - so maybe we could just expand the config interface:
https://ionicframework.com/docs/api/modal#passing-parameters

### PR DESCRIPTION
**Pull request by hafstad**
closes #261 
image

### CODE REVIEW

File: src/app/showcase/modal-showcase/modal-showcase.component.html [hunk 1]
```diff
@@ -25,22 +29,32 @@ <h3>Inside the parent (caller) component:</h3>
 {{ '}' }}</kirby-code-viewer>
     </p>
     <br />
-    <h3>Inside the nested component:</h3>
-    <p>To create a nested component, ensure:</p>
+    <h3>Inside the embeded component:</h3>
```
Comment (jakobe) [inline src/app/showcase/modal-showcase/modal-showcase.component.html]:
```suggestion
    <h3>Inside the embedded component:</h3>
```

File: src/app/showcase/modal-showcase/modal-showcase.component.html [hunk 2]
```diff
@@ -25,22 +29,32 @@ <h3>Inside the parent (caller) component:</h3>
...
+    <p>To create a embeded component, ensure:</p>
```
Comment (jakobe) [inline src/app/showcase/modal-showcase/modal-showcase.component.html]:
```suggestion
    <p>To create a embedded component, ensure:</p>
```

File: src/app/showcase/modal-showcase/modal-showcase.component.html [hunk 3]
```diff
@@ -25,22 +29,32 @@ <h3>Inside the parent (caller) component:</h3>
...
     <ol>
-        <li>the nested component must be declared (module file, or inside your <code>@Component</code> decorator)</li>
-        <li>the nested component must be listed in your <code>entryComponents</code> (module file)</li>
+        <li>the embeded component must be declared (module file, or inside your <code>@Component</code> decorator)</li>
```
Comment (jakobe) [inline src/app/showcase/modal-showcase/modal-showcase.component.html]:
```suggestion
        <li>the embedded component must be declared (module file, or inside your <code>@Component</code> decorator)</li>
```

File: src/app/showcase/modal-showcase/modal-showcase.component.html [hunk 4]
```diff
@@ -25,22 +29,32 @@ <h3>Inside the parent (caller) component:</h3>
...
+        <li>the embeded component must be listed in your <code>entryComponents</code> (module file)</li>
```
Comment (jakobe) [inline src/app/showcase/modal-showcase/modal-showcase.component.html]:
```suggestion
        <li>the embedded component must be listed in your <code>entryComponents</code> (module file)</li>
```

File: src/app/showcase/modal-showcase/modal-showcase.component.html [hunk 5]
```diff
@@ -25,22 +29,32 @@ <h3>Inside the parent (caller) component:</h3>
...
+    <kirby-code-viewer language="ts">constructor(
+    @Inject(COMPONENT_PROPS) private componentProps,
+    private modalController: ModalController,
+    private vcRef: ViewContainerRef
+) {{ '{' }}
+    this.props = componentProps;
+{{ '}' }}</kirby-code-viewer><br />
     <p>
-        In order to hide a modal from within your nested component (and destroy its component), you can use:
+        In order to hide a modal from within your embeded component (and destroy its component), you can use:
```
Comment (jakobe) [inline src/app/showcase/modal-showcase/modal-showcase.component.html]:
```suggestion
        In order to hide a modal from within your embedded component (and destroy its component), you can use:
```

File: src/app/showcase/modal-showcase/modal-showcase.component.html [hunk 6]
```diff
@@ -25,22 +29,32 @@ <h3>Inside the parent (caller) component:</h3>
...
         <kirby-code-viewer language="ts">constructor(private modalController: ModalController) {{ '{}' }}
 this.modalController.hideTopmost();</kirby-code-viewer>
         <br /><br />
         <em>(Optional)</em><br />
         You can additionally pass data, which will be available in your callback:
         <kirby-code-viewer language="ts">let returnData = ...;
 this.modalController.hideTopmost(returnData);</kirby-code-viewer>
-        It is possible to nest more modals from within your nested components, <br />
+        It is possible to nest more modals from within your embeded components, <br />
```
Comment (jakobe) [inline src/app/showcase/modal-showcase/modal-showcase.component.html]:
```suggestion
        It is possible to nest more modals from within your embedded components, <br />
```

File: src/app/showcase/modal-showcase/modal-showcase.component.html [hunk 7]
```diff
@@ -25,22 +29,32 @@ <h3>Inside the parent (caller) component:</h3>
...
+    <p>To create a embeded component, ensure:</p>
     <ol>
-        <li>the nested component must be declared (module file, or inside your <code>@Component</code> decorator)</li>
-        <li>the nested component must be listed in your <code>entryComponents</code> (module file)</li>
+        <li>the embeded component must be declared (module file, or inside your <code>@Component</code> decorator)</li>
+        <li>the embeded component must be listed in your <code>entryComponents</code> (module file)</li>
     </ol>
+    <br />
+    <em>(Optional)</em><br />
+    <p>To access data passed into the embeded component, inject the COMPONENT_PROPS inside the embeded components constructor:</p>
```
Comment (jakobe) [inline src/app/showcase/modal-showcase/modal-showcase.component.html]:
```suggestion
    <p>To access data passed into the embedded component, inject the COMPONENT_PROPS inside the embedded components constructor:</p>
```

File: src/app/examples/modal-example/first-embedded-modal-example/first-embedded-modal-example.component.ts [hunk 8]
```diff
@@ -1,14 +1,23 @@
...
 import { ModalWrapperConfig } from '~/kirby/components/modal/modal-wrapper/config/modal-wrapper-config';
 import { SecondEmbeddedModalExampleComponent } from '../second-embedded-modal-example/second-embedded-modal-example.component';
+import { COMPONENT_PROPS } from '~/kirby/components/modal/modal-wrapper/config/modal-wrapper-config.helper';
 
 @Component({
   templateUrl: './first-embedded-modal-example.component.html',
 })
 export class FirstEmbeddedModalExampleComponent {
-  constructor(private modalController: ModalController, private vcRef: ViewContainerRef) {}
+  props: object;
```
Comment (jakobe) [inline src/app/examples/modal-example/first-embedded-modal-example/first-embedded-modal-example.component.ts]:
```suggestion
  props: { [key: string]: any };
```

File: src/app/showcase/modal-showcase/modal-showcase.component.ts [hunk 9]
```diff
@@ -43,5 +43,11 @@ export class ModalShowcaseComponent {
       defaultValue: '',
       inputValues: ['Component'],
     },
+    {
+      name: 'componentProps',
+      description: 'The data to pass to the modal component.',
+      defaultValue: '',
+      inputValues: ['undefined | { [key: string]: any; }'],
```
Comment (mmmtrifork) [inline src/app/showcase/modal-showcase/modal-showcase.component.ts]: Isn't the the undefined type redundant?

File: src/kirby/components/modal/modal-wrapper/config/modal-wrapper-config.ts [hunk 10]
```diff
@@ -4,4 +4,5 @@ export interface ModalWrapperConfig {
   closeIconName?: 'close' | 'arrow-back';
   dim?: number;
   component: any;
+  componentProps?: undefined | { [key: string]: any };
```
Comment (mmmtrifork) [inline src/kirby/components/modal/modal-wrapper/config/modal-wrapper-config.ts]: Isn't the the undefined type redundant?

File: src/app/showcase/modal-showcase/modal-showcase.component.ts (hunk 9 above)
Comment (hafstad) [reply src/app/showcase/modal-showcase/modal-showcase.component.ts]: You can argue that it is, however I copied this from Ionic's documentation and personally it indicates for me that it's "optional".

Comment (hafstad) [reply src/app/showcase/modal-showcase/modal-showcase.component.ts]: You can argue that it is, however I copied this from Ionic's documentation and personally it indicates for me that it's "optional".

**Pull request merged**